### DISCUSSION TITLE
shape intersection with complex object fails

### DISCUSSION THREAD

**Discussion by tve** [sgenoud/replicad#98] [Modelling help]
I have a complex object and am trying to cut out a section so I can print just that to test the fit, but the intersection fails. Here is a screen shot that shows the object 3 times:
- the colorful version to the left shows the individual parts that are fused
- the light grey version to the right is the fused final object
- in the center is the final object in light green and a box I want to intersect with in light blue

![image](https://github.com/sgenoud/replicad/assets/39480/8563c5c9-b673-4b46-94ce-f94ee9fe0c6b)

But when I perform the intersection I get:

![image](https://github.com/sgenoud/replicad/assets/39480/5f2ea60a-73dc-4f27-afbf-fd0ba41f61a3)

Somehow this cylinder piece is the intersection between the box and one of the cylinders that make up the complex object. Everything else is missing...

Model source: https://studio.replicad.xyz/workbench?code=UEsDBAoAAAAIABiL9lbruTHXfwUAAIQPAAAHAAAAY29kZS5qc41XW2%252FbNhR%252B96848F6kVJYdF8HWAh2WS7s9bGjQpkvaog%252BMRFlaaFKlqDhGkB%252B0v7FftsObRMl2tiCGeTnX73yHpOdz%252BFRvKp6ftXm%252BBfLAKKxFyxUUQgJh7bri7RrWpFEJ%252FHk8mc%252FhXNRbWa1KBf%252F8vVwsXyZwVQrZKMrhXnB4W2V3lE8m8yNYMXFLGEhasyojORzNcfUIflHbmua0gEfQI1FAta6FVNHUS05jeBqoTTLBGwUS3nTLfq3IFK4uJ%252FRB2wC7iNZJy9QlkWTd4PbjBABDX5OKmxxxqr8uKvIajl8lbvo75ThdLGCurfrVa8LYa3iZWBMl4bnRtwOj8mOgYZcvRRMsa71aNJWqEB%252FCBF%252F5IKz0KV8x%252Bhp%252BOjGSRM9ACWgYwTpYEWjrDZF5Y4NoGKX3RqYULO%252FLRB%252BUbBv0gmJWxsZ30sdnl21O1p8qsWKcNg1gJTa4AUQiA3L4BLUURWUCba%252BrXJUandRqbfRca0S3QimxxmE81LigtdF4aeRLaiijFZoqp82u%252FJWOA%252BXTPWE5MROex0BJyleqpLxCPHGXUTVXsjLwoQgOXT1d5ZpM0o3GjInsTmM2hEvvGj6cJH76vij6eERRNBTpJjHXjHJF5eRpMilanpmyZlhWeiol2UYkRmZLqlqJ1U7XpI4ogzc%252FA2WpkYpipPfEE9ZRFTpLmqRRHRvOWjabrkTmo607er5l2K5URnXqKIzFXSZgp5hynKIigBSKKBq9WiTwFT%252F6%252FxsOZ8duHKc6K2YwBGyLz%252BimaoyqkoQj%252BRT9MnQSGw2yBcERxpvPUCND6aSL01G1i1RgWyORogCbr8aBb0WnkLmUbNz%252FkWXXdy7PLlO%252FZbppmDR%252Bm7x3kp7dmKyRk0zTk21t09Hc91vsfByEJNmPicuyaWtd5Pkd3W7INsjvjDT0TDxEQT6%252BhKOKekfpOJJoFihbDZ1vnHjnd1zc%252BrH1%252BrEuqaR9BkfYzSfOshHKJdmcVzJju0LNHVVZ%252BZ5f6vyiOJX0XrB7Gn09ThaJRnZo5VKwrWn0kaEElov%252Fa%252BygJTiC3lbWqshKfqCZMu3fSwYoxmkPnSbEEN5nY%252FId82wJjNTAKLwI%252BYpB%252F0FUmWaiGXIVN2R68fbX5YfTi1gT7lvsXHWOPncaeLMcbPCBc8d7S3tTFyMPjfBtV4uKqwaZ3newvR3MyY1tXKf2EMc8lhhknfaXx0jD3A5Www6f09BH629i%252F5Hmj%252BE5sjkw%252FWI5rl6wOV%252B%252BcIp4YOtOj0fRdY76rutuxyTMeeDTwWxvlsg3oGabP3cO2cTsj5MOvaRDpSNQSCFfuh4pe%252BbsUmAWVqc%252FEnREHaZa11x3Nn%252B8BHV8CADHu0%252Bm70hG31ms47RyVJ%252FefJnqgyMtcCeyThJ4bHn1vcVnCd6R9KmHtDd5sOmwHPb2PRpWcTaq4qCG4wa0US0Hjq9E3fkdH1MBIgNwRwKHj%252Bt97mc%252Bk0EYGIMNJmWiUNeVKiOHisbQ0AU5aF8dWBjddQgtvoTF7V%252BIVn%252Bv41vr2ctSR564dvVIJto3nhBD%252B4pqtlP7gsDMspLiI6eoFD7UkC32hViNnoguIVT9iJpI4p1GWS4G%252FWEONAuI5vhiUF2zsq8CsxP9iQcOT03qCIB%252FEoWkdQENBZCu%252BObSKUZBxN2LaicdXaRducBKYNuA6V5s9oXyCE1JamS%252FluqyuYnw1wFe%252BZyscWuKe8iRTDAhcfZDUVD8m8JTMjKBVQxtzIZGcDewIlButceIJcFhM3Y%252FMHTL2j1mbL0Om7H7gZmVpJTvseMPnGdMeZHA2pYyJjbTsTWk1GE7uBniw6r7PXn5UvdabiFQrVtZs2d0kSY76rgWWMAfuZTtB9aResfA6YAj5pUZQvoNf0T8C1BLAQIUAAoAAAAIABiL9lbruTHXfwUAAIQPAAAHAAAAAAAAAAAAAAAAAAAAAABjb2RlLmpzUEsFBgAAAAABAAEANQAAAKQFAAAAAA%253D%253D

To see the result of the intersection select only the "testSect" object in the viewer.

**Comment by sgenoud** [comment]
There was two problems with your code! The main one is that you used `makeCompound` when you should have used some fusing. `makeCompound` keeps the different parts separated - but you want to fuse them into one. When you do an intersect on a compounded shape it gets confused.

There is also an issue with your knob - when you revolve your shape it creates some weird self intersections. It works with the sphere for instance.

Here is the [code with my modifications](https://studio.replicad.xyz/workbench?code=UEsDBAoAAAAIADVEDlfKwpALpQUAALYPAAAHAAAAY29kZS5qc41X3W7bNhS%252B91MceDdSKsuOg2BrgQxLm3a72NCgP2vaIBeMdGRpoUWVpOIYQR5or7En2yGpH0q2swUxLJHn9zvfOaTnc%252FhcbYoyfV2n6RbYA0dYi7rUkAkJjNfroqzXsGZKR%252FDn8WQ%252Bhzei2spilWv45%252B%252FlYnkSwadcSKWxhHtRwtsiucNyMpkfwYqLW8ZBYsWLhKVwNKfVI%252FhFbytMMYNHME8ig2JdCamDaSs5DeFpoDZJRKk0SDjrltu1LNG0upzgg7EBbpGss5rrSybZWtH24wSAQl%252BzorQ50qv5uijYKzh%252BGTWvv2NJr4sFzI3VdvUL4%252FwVnETORM7K1Oq7B6vyo6fhli%252BF8paNXiVUoQvCh3FRrtognPR5ueL4Cn46tZLMvIEWoDijOjgRqKsNk6lyQSiOeG9lcsHTvkz4oGWtyAuJORkX32kfn1t2OTl%252FOqeKlagUUCU2tAFMEgNS%252BAyVFFlhA62%252FFKnODTqx09qYd6MR3AqtxZoew6HGBVZW48TK52gpYxRUkaLalf9k4iD5eE9YjZgNr8VASyxXOseyIDxpl6Oea1lY%252BEiEHpt6NpVTicSNwYyL5M5gNoTL7Fo%252BnEbt6%252Fss6%252BMRWaaQ6CYp1wRLjXLyNJlkdZnYsiZUVjyXkm0DFhKzJepaUrXjNasC5HD2MyCPrVQQEr07%252BtYKzwnzMzB6JMViRfzG4DiMJaY1PQUsSSJQbjdJYqMSqM5WBOx6cdO9TtpOaHoAuhAN%252B4MqtM3gnNt2p5aiIO%252FwzZbTHEAZVHHTG8SaZQTulbAMY1IEkEIzjcHLRQTX9DH%252FN%252FQ4O26ew9jAxW1xgPrtK7kplFXVkpXEao3fhk5Cq8G2IEqqz9VXqIj6OOnibHrgrAEruLbW2oZudpMmfhfkf6TUdW%252BTVJdWu2V7cpghfdskdzKcXdkUidnckJxvXeti2nZt2Pg4mH%252B0H4AmS1VXpqLzO9xu2NbL7zVT%252BFo8BF4%252Bbb1G5WsdxeNIgpmn7DRMvmHUOr8rxa1Tcj4%252FVjlK7OM%252Foolw2tglcRmnkm3eFDLhu0LqDnWSvy8vTXaBIfi94PcYXB9Hi8jgOrRyKfjWDouRoQiWi%252F9v7KAtOILeWlLrwEl%252BwETbIdJLeiiGcQ%252BdIcQQ3mejatvj2RJYqYFReOHzlYL%252Bg%252Bk8ToQacpU2ZHzx9tflh%252FOL0BCu9dT5%252Bdop0PF0sJkHvhvaO9bbwlh5UKLtukoUpVZE9L5b3RFjxz8Nlyp2JwGlsaQYq7g%252FgUYa9ohxGu7xOQ0zn38T%252B8dXO8vnRGbP9IvluHje5nz5olGkqW8aPRxF1znqm647YiM%252F54HPBmZ3PAVt%252FxmytWPnkE3K%252Fjjq0Is6VDr%252B%252BAxqS9cj5UbOLgVmfnX6iWAi6jA1uvbMdPnTSWriIwBKOkBl%252FI4l%252BM5hHcZFw%252FTp1bepmRtxRjuBcxLBY10W32u629BBi089pL3Jgz1H5XBH%252BNGwirNRFQc1HPefi2o5cPxJVJ3f8ZzyEBmAOxI4PK33uZ%252B1mQzCoBhcMDEXmf5S6DxoUDEYWroQB93VhQpjuo6gpeu0uP2L0OrPcHt56E5GE2fUNGeLW2Q83YyMaTTURnc1oDSSHOlalBWarnZEDXenLEaXyiZ6Uv1ImsTYna5YLgbNYIeXy94QejEopV3ZB%252Ffs1HzCgUN3SaJs25uOz9AmoKEAcZNuaSbFwIu4vyiN0zEV2ZXzrHi2LZjNHc%252FdRh5B5awiqhupLpurgH5P0PFesjVtTWmPCJEILiS9%252FZBlSH9TeIpGJqiIvo3Z0AjtelYEya32GHEcOGzG7XuGbnm9x4yr12Ezbt8zs5KI5R477XR5xlQr4lnbIudiMx1bI0odtkObPj68uN%252BTV1vqXqtZ8FSrWlb8GV2iyY46rXkW6Gcx8v3ANqTeMXA%252B4Ii9UfqQ3tDPjn8BUEsBAhQACgAAAAgANUQOV8rCkAulBQAAtg8AAAcAAAAAAAAAAAAAAAAAAAAAAGNvZGUuanNQSwUGAAAAAAEAAQA1AAAAygUAAAAA).
